{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "from Bio import AlignIO\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['.ipynb_checkpoints',\n",
       " 'antigens_supp_dat.xlsx',\n",
       " 'indels_filter_summary.csv',\n",
       " 'indels_passes_all.csv',\n",
       " 'isovar_indels_results.csv',\n",
       " 'Isovar_results_summary.ipynb',\n",
       " 'isovar_snvs_results.csv',\n",
       " 'snvs_filter_summary.csv',\n",
       " 'snvs_passes_all.csv']"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = os.listdir(os.getcwd())\n",
    "\n",
    "#dfs = [pd.read_csv(i) for i in data if \".csv\" in i]\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "indels = dfs[0]\n",
    "snvs = dfs[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "def summarize_isovar(df):\n",
    "    filt = df[[i for i in df.columns if \"filter\" in i]]\n",
    "    new_df = pd.DataFrame()\n",
    "    for col in filt.columns:\n",
    "        new_df[col] = filt[col].value_counts()\n",
    "        \n",
    "    new_df = new_df.T.fillna(0)\n",
    "    new_df[\"percent_true\"] = np.round((new_df[True] / (new_df[True] + new_df[False])) * 100, 2)\n",
    "    \n",
    "    return new_df\n",
    "\n",
    "summarize_isovar(snvs).to_csv(\"snvs_filter_summary.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "def batch_read_csv(file_paths, keyword):\n",
    "    return [pd.read_csv(i) for i in file_paths if keyword in i]\n",
    "\n",
    "summaries = batch_read_csv(data, \"summary\")\n",
    "passes_all = batch_read_csv(data, \"passes_all\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "s_pass = pd.read_csv(\"snvs_passes_all.csv\", index_col=\"Unnamed: 0\")\n",
    "i_pass = pd.read_csv(\"indels_passes_all.csv\", index_col=\"Unnamed: 0\")\n",
    "anti = pd.read_excel(\"antigens_supp_dat.xlsx\")\n",
    "s_all = pd.read_csv(\"isovar_snvs_results.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "bbn963 = anti.groupby(\"Tumor_model\").get_group(\"BBN963\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "s_all_dropped = s_all.loc[s_all_dropped.predicted_effect_mutant_protein_sequence.dropna().index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "def match_seq(df, antis, col):\n",
    "    dic = {}\n",
    "    idx = []\n",
    "    df = pd.DataFrame()\n",
    "    for a in antis:\n",
    "        for i in df.index:\n",
    "            if a in df[col].loc[i]:\n",
    "                dic[a] = (i, df[col].loc[i])\n",
    "                idx.append(i)\n",
    "                \n",
    "    return df.loc[set(idx)], dic\n",
    "            \n",
    "            \n",
    "\n",
    "match_rna, rna_dic = match_seq(s_pass, bbn963.Antigen, \"protein_sequence\")\n",
    "match_dna, dna_dic = match_seq(s_pass, bbn963.Antigen, \"predicted_effect_mutant_protein_sequence\")\n",
    "#all_dict = match_seq(s_all_dropped, bbn963.Antigen, \"predicted_effect_mutant_protein_sequence\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def summarize_matches(df, antis):\n",
    "\n",
    "    summ_df = pd.DataFrame(index=antis.values, columns=[\"match_index\", \"protein_sequence\", \"predicted_effect_mutant_protein_sequence\", \"Both\", \"passes_all_filters\"])\n",
    "    \n",
    "    for a in antis:\n",
    "        for i in df.index:\n",
    "            \n",
    "            try:\n",
    "                if a in df.loc[i, 'protein_sequence']:\n",
    "                    summ_df.loc[a, \"passes_all_filters\"] = df.loc[i, 'passes_all_filters']\n",
    "                    summ_df.loc[a, \"match_index\"] = i\n",
    "                    summ_df.loc[a, 'protein_sequence'] = True\n",
    "            except TypeError:\n",
    "                pass\n",
    "            \n",
    "            try:\n",
    "                if a in df.loc[i, \"predicted_effect_mutant_protein_sequence\"]:\n",
    "                    summ_df.loc[a, \"passes_all_filters\"] = df.loc[i, 'passes_all_filters']\n",
    "                    summ_df.loc[a, \"match_index\"] = i\n",
    "                    summ_df.loc[a, 'predicted_effect_mutant_protein_sequence'] = True\n",
    "            except TypeError:\n",
    "                pass\n",
    "    for i in summ_df.index:\n",
    "        if summ_df.loc[i, 'protein_sequence'] != True:\n",
    "            summ_df.loc[i, ['protein_sequence', 'Both']] = False\n",
    "            \n",
    "        if summ_df.loc[i, 'predicted_effect_mutant_protein_sequence'] != True:\n",
    "            summ_df.loc[i, ['predicted_effect_mutant_protein_sequence', 'Both']] = False\n",
    "    \n",
    "        \n",
    "    summ_df.Both = summ_df.Both.fillna(True)\n",
    "    \n",
    "    return summ_df\n",
    "\n",
    "summ = summarize_matches(s_all, bbn963.Antigen)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>match_index</th>\n",
       "      <th>protein_sequence</th>\n",
       "      <th>predicted_effect_mutant_protein_sequence</th>\n",
       "      <th>Both</th>\n",
       "      <th>passes_all_filters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>FSLNVVNL</th>\n",
       "      <td>7292</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FTFTSFWAYKI</th>\n",
       "      <td>9926</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GMAVRALNM</th>\n",
       "      <td>4530</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IAICVSDTV</th>\n",
       "      <td>18133</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ISRNRHTL</th>\n",
       "      <td>16309</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SMGSNDSDTL</th>\n",
       "      <td>12136</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YFLENLDKL</th>\n",
       "      <td>3791</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ETLLNSATI</th>\n",
       "      <td>2683</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LMMLKNVGFAL</th>\n",
       "      <td>14271</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNVMQLLL</th>\n",
       "      <td>9800</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATVIYVPPPFAAAAI</th>\n",
       "      <td>15941</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DEEIFSIFAPTPTAM</th>\n",
       "      <td>2709</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FTPLILAATAGHAGV</th>\n",
       "      <td>9103</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IIGGLSEASPATSEI</th>\n",
       "      <td>7501</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                match_index protein_sequence  \\\n",
       "FSLNVVNL               7292             True   \n",
       "FTFTSFWAYKI            9926             True   \n",
       "GMAVRALNM              4530             True   \n",
       "IAICVSDTV             18133             True   \n",
       "ISRNRHTL              16309             True   \n",
       "SMGSNDSDTL            12136            False   \n",
       "YFLENLDKL              3791             True   \n",
       "ETLLNSATI              2683             True   \n",
       "LMMLKNVGFAL           14271             True   \n",
       "SNVMQLLL               9800             True   \n",
       "ATVIYVPPPFAAAAI       15941            False   \n",
       "DEEIFSIFAPTPTAM        2709            False   \n",
       "FTPLILAATAGHAGV        9103            False   \n",
       "IIGGLSEASPATSEI        7501             True   \n",
       "\n",
       "                predicted_effect_mutant_protein_sequence   Both  \\\n",
       "FSLNVVNL                                            True   True   \n",
       "FTFTSFWAYKI                                         True   True   \n",
       "GMAVRALNM                                           True   True   \n",
       "IAICVSDTV                                           True   True   \n",
       "ISRNRHTL                                            True   True   \n",
       "SMGSNDSDTL                                          True  False   \n",
       "YFLENLDKL                                           True   True   \n",
       "ETLLNSATI                                           True   True   \n",
       "LMMLKNVGFAL                                         True   True   \n",
       "SNVMQLLL                                            True   True   \n",
       "ATVIYVPPPFAAAAI                                     True  False   \n",
       "DEEIFSIFAPTPTAM                                     True  False   \n",
       "FTPLILAATAGHAGV                                     True  False   \n",
       "IIGGLSEASPATSEI                                     True   True   \n",
       "\n",
       "                passes_all_filters  \n",
       "FSLNVVNL                      True  \n",
       "FTFTSFWAYKI                   True  \n",
       "GMAVRALNM                     True  \n",
       "IAICVSDTV                     True  \n",
       "ISRNRHTL                      True  \n",
       "SMGSNDSDTL                   False  \n",
       "YFLENLDKL                     True  \n",
       "ETLLNSATI                     True  \n",
       "LMMLKNVGFAL                   True  \n",
       "SNVMQLLL                     False  \n",
       "ATVIYVPPPFAAAAI               True  \n",
       "DEEIFSIFAPTPTAM               True  \n",
       "FTPLILAATAGHAGV               True  \n",
       "IIGGLSEASPATSEI               True  "
      ]
     },
     "execution_count": 243,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matches = summ.dropna().drop_duplicates(\"match_index\")\n",
    "\n",
    "\n",
    "#matches.to_csv(\"isovar_paper_matches.csv\")\n",
    "\n",
    "counts = pd.DataFrame(matches.iloc[:, 1:].astype(bool).sum(), columns=[True])#, False, \"percent\"])\n",
    "counts[False] = 14 - counts\n",
    "\n",
    "\n",
    "matches\n",
    "#matches.reset_index().rename(columns={\"index\":\"Antigen\"}).to_csv(\"isovar_paper_matches.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Antigen</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Predicted_MHC_class</th>\n",
       "      <th>Tumor_model</th>\n",
       "      <th>MHC_haplotype</th>\n",
       "      <th>ELISpot_score</th>\n",
       "      <th>Group</th>\n",
       "      <th>GBM_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>FSLNVVNL</td>\n",
       "      <td>FSPNVVNL</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Kb</td>\n",
       "      <td>-51.333333</td>\n",
       "      <td>Exploration</td>\n",
       "      <td>1.933884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>FSLNVVNLTLV</td>\n",
       "      <td>FSPNVVNLTLV</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Db</td>\n",
       "      <td>283.333333</td>\n",
       "      <td>Exploration</td>\n",
       "      <td>39.644072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>FTFTSFWAYKI</td>\n",
       "      <td>FIFTSFWAYKI</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Kb</td>\n",
       "      <td>-23.166700</td>\n",
       "      <td>Exploration</td>\n",
       "      <td>34.773693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GMAVRALNM</td>\n",
       "      <td>GMAVRELNM</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Kb</td>\n",
       "      <td>23.833333</td>\n",
       "      <td>Exploration</td>\n",
       "      <td>34.773693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>IAICVSDTV</td>\n",
       "      <td>IAICVSVTV</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Db</td>\n",
       "      <td>33.000000</td>\n",
       "      <td>Exploration</td>\n",
       "      <td>43.318294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>ISRNRHTL</td>\n",
       "      <td>ISCNRHTL</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Kb</td>\n",
       "      <td>13.833332</td>\n",
       "      <td>Exploration</td>\n",
       "      <td>1.933884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>VMISRNRHTL</td>\n",
       "      <td>VMISCNRHTL</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Kb</td>\n",
       "      <td>12.000000</td>\n",
       "      <td>Exploration</td>\n",
       "      <td>37.532123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>VVMISRNRHTL</td>\n",
       "      <td>VVMISCNRHTL</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Kb</td>\n",
       "      <td>-7.833330</td>\n",
       "      <td>Exploration</td>\n",
       "      <td>37.532123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>YFLENLDKL</td>\n",
       "      <td>YFLDNLDKL</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Db</td>\n",
       "      <td>-18.166700</td>\n",
       "      <td>Exploration</td>\n",
       "      <td>1.933884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>ETLLNSATI</td>\n",
       "      <td>ETLLNSATT</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Db</td>\n",
       "      <td>-41.333333</td>\n",
       "      <td>Validation</td>\n",
       "      <td>1.933884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>KVFSLNVVNL</td>\n",
       "      <td>KVFSPNVVNL</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Kb</td>\n",
       "      <td>7.666667</td>\n",
       "      <td>Validation</td>\n",
       "      <td>22.440171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>LMMLKNVGFAL</td>\n",
       "      <td>LLMLKNVGFAL</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Kb</td>\n",
       "      <td>68.000000</td>\n",
       "      <td>Validation</td>\n",
       "      <td>13.169980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>MISRNRHTL</td>\n",
       "      <td>MISCNRHTL</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Db</td>\n",
       "      <td>25.999998</td>\n",
       "      <td>Validation</td>\n",
       "      <td>1.933884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>MMLKNVGFAL</td>\n",
       "      <td>LMLKNVGFAL</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Db</td>\n",
       "      <td>-18.166700</td>\n",
       "      <td>Validation</td>\n",
       "      <td>13.169980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>MYFTFTSF</td>\n",
       "      <td>MYFIFTSF</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Kb</td>\n",
       "      <td>13.000001</td>\n",
       "      <td>Validation</td>\n",
       "      <td>34.773693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>RALNMNLL</td>\n",
       "      <td>RELNMNLL</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Db</td>\n",
       "      <td>30.166663</td>\n",
       "      <td>Validation</td>\n",
       "      <td>34.773693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>SIFIEMYFTF</td>\n",
       "      <td>SIFIEMYFIF</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Kb</td>\n",
       "      <td>112.000000</td>\n",
       "      <td>Validation</td>\n",
       "      <td>34.773693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>SNVMQLLL</td>\n",
       "      <td>SDVMQLLL</td>\n",
       "      <td>I</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-Kb</td>\n",
       "      <td>-37.000000</td>\n",
       "      <td>Validation</td>\n",
       "      <td>54.975770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>IIGGLSEASPATSEI</td>\n",
       "      <td>IIGGLSEARPATSEI</td>\n",
       "      <td>II</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-IA</td>\n",
       "      <td>5.833337</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>LILAATAGHAGVVEI</td>\n",
       "      <td>LILAATAGHVGVVEI</td>\n",
       "      <td>II</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-IA</td>\n",
       "      <td>-6.166654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>PLILAATAGHAGVVE</td>\n",
       "      <td>PLILAATAGHVGVVE</td>\n",
       "      <td>II</td>\n",
       "      <td>BBN963</td>\n",
       "      <td>H2-IA</td>\n",
       "      <td>-6.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Antigen        Reference Predicted_MHC_class Tumor_model  \\\n",
       "19          FSLNVVNL         FSPNVVNL                   I      BBN963   \n",
       "20       FSLNVVNLTLV      FSPNVVNLTLV                   I      BBN963   \n",
       "21       FTFTSFWAYKI      FIFTSFWAYKI                   I      BBN963   \n",
       "24         GMAVRALNM        GMAVRELNM                   I      BBN963   \n",
       "35         IAICVSDTV        IAICVSVTV                   I      BBN963   \n",
       "38          ISRNRHTL         ISCNRHTL                   I      BBN963   \n",
       "125       VMISRNRHTL       VMISCNRHTL                   I      BBN963   \n",
       "130      VVMISRNRHTL      VVMISCNRHTL                   I      BBN963   \n",
       "136        YFLENLDKL        YFLDNLDKL                   I      BBN963   \n",
       "148        ETLLNSATI        ETLLNSATT                   I      BBN963   \n",
       "159       KVFSLNVVNL       KVFSPNVVNL                   I      BBN963   \n",
       "162      LMMLKNVGFAL      LLMLKNVGFAL                   I      BBN963   \n",
       "166        MISRNRHTL        MISCNRHTL                   I      BBN963   \n",
       "167       MMLKNVGFAL       LMLKNVGFAL                   I      BBN963   \n",
       "168         MYFTFTSF         MYFIFTSF                   I      BBN963   \n",
       "173         RALNMNLL         RELNMNLL                   I      BBN963   \n",
       "181       SIFIEMYFTF       SIFIEMYFIF                   I      BBN963   \n",
       "182         SNVMQLLL         SDVMQLLL                   I      BBN963   \n",
       "236  IIGGLSEASPATSEI  IIGGLSEARPATSEI                  II      BBN963   \n",
       "241  LILAATAGHAGVVEI  LILAATAGHVGVVEI                  II      BBN963   \n",
       "251  PLILAATAGHAGVVE  PLILAATAGHVGVVE                  II      BBN963   \n",
       "\n",
       "    MHC_haplotype  ELISpot_score        Group  GBM_score  \n",
       "19          H2-Kb     -51.333333  Exploration   1.933884  \n",
       "20          H2-Db     283.333333  Exploration  39.644072  \n",
       "21          H2-Kb     -23.166700  Exploration  34.773693  \n",
       "24          H2-Kb      23.833333  Exploration  34.773693  \n",
       "35          H2-Db      33.000000  Exploration  43.318294  \n",
       "38          H2-Kb      13.833332  Exploration   1.933884  \n",
       "125         H2-Kb      12.000000  Exploration  37.532123  \n",
       "130         H2-Kb      -7.833330  Exploration  37.532123  \n",
       "136         H2-Db     -18.166700  Exploration   1.933884  \n",
       "148         H2-Db     -41.333333   Validation   1.933884  \n",
       "159         H2-Kb       7.666667   Validation  22.440171  \n",
       "162         H2-Kb      68.000000   Validation  13.169980  \n",
       "166         H2-Db      25.999998   Validation   1.933884  \n",
       "167         H2-Db     -18.166700   Validation  13.169980  \n",
       "168         H2-Kb      13.000001   Validation  34.773693  \n",
       "173         H2-Db      30.166663   Validation  34.773693  \n",
       "181         H2-Kb     112.000000   Validation  34.773693  \n",
       "182         H2-Kb     -37.000000   Validation  54.975770  \n",
       "236         H2-IA       5.833337          NaN        NaN  \n",
       "241         H2-IA      -6.166654          NaN        NaN  \n",
       "251         H2-IA      -6.000000          NaN        NaN  "
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bbn963.loc[bbn963.Antigen.isin(list(s_dict.keys()))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "568    AAELFGLTAEKIYLVHDELD\n",
       "299    AAREAAKTVKTKCGSKLFPL\n",
       "728          AASGSSASSIHFSS\n",
       "439    AASQRRDVERELTRFMAKTG\n",
       "901    AATSNTTSCSNPVAASKMST\n",
       "745    ACEMVADMVETLQSVLALGH\n",
       "938        ADKTGDIIVTNDNFRE\n",
       "384    ADLLPPPPAQPPPHSNSEEY\n",
       "197    AEAGYTSRGKSGCTQPRRVA\n",
       "386    AEEHQRVSKGNLKAFGNSTE\n",
       "Name: protein_sequence, dtype: object"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s_pass.protein_sequence.sort_values().iloc[0:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['variant', 'overlapping_gene_names', 'overlapping_gene_ids',\n",
       "       'overlapping_coding_gene_names', 'overlapping_coding_gene_ids',\n",
       "       'fraction_alt_fragments', 'fraction_alt_reads',\n",
       "       'fraction_other_fragments', 'fraction_other_reads',\n",
       "       'fraction_ref_fragments', 'fraction_ref_reads', 'num_alt_fragments',\n",
       "       'num_alt_reads', 'num_amino_acid_mismatches_from_predicted_effect',\n",
       "       'num_amino_acid_mismatches_from_reference',\n",
       "       'num_cdna_mismatches_after_variant_in_top_protein_sequence',\n",
       "       'num_cdna_mismatches_before_variant_in_top_protein_sequence',\n",
       "       'num_cdna_mismatches_in_top_protein_sequence',\n",
       "       'num_cdna_sequences_from_top_protein_sequence',\n",
       "       'num_fragments_supporting_top_protein_sequence',\n",
       "       'num_genes_from_protein_sequences',\n",
       "       'num_genes_from_top_protein_sequence',\n",
       "       'num_mutant_amino_acids_in_protein_sequence', 'num_other_fragments',\n",
       "       'num_other_reads', 'num_overlapping_coding_genes',\n",
       "       'num_overlapping_coding_transcripts', 'num_overlapping_genes',\n",
       "       'num_overlapping_transcripts', 'num_phased_variants_from_reads',\n",
       "       'num_phased_variants_in_protein_sequence', 'num_protein_sequences',\n",
       "       'num_reads_supporting_top_protein_sequence', 'num_ref_fragments',\n",
       "       'num_ref_reads', 'num_total_fragments', 'num_total_reads',\n",
       "       'num_transcripts_from_protein_sequences',\n",
       "       'num_transcripts_from_top_protein_sequence',\n",
       "       'num_translations_from_top_protein_sequence',\n",
       "       'ratio_alt_to_other_fragments', 'ratio_alt_to_other_reads',\n",
       "       'ratio_other_to_alt_fragments', 'ratio_other_to_alt_reads',\n",
       "       'ratio_other_to_ref_fragments', 'ratio_other_to_ref_reads',\n",
       "       'ratio_ref_to_other_fragments', 'ratio_ref_to_other_reads',\n",
       "       'has_mutant_protein_sequence_from_predicted_effect',\n",
       "       'has_mutant_protein_sequence_from_rna',\n",
       "       'has_reference_protein_sequence_from_predicted_effect',\n",
       "       'predicted_effect', 'predicted_effect_class',\n",
       "       'predicted_effect_modifies_protein_sequence',\n",
       "       'predicted_effect_gene_name', 'predicted_effect_gene_id',\n",
       "       'predicted_effect_transcript_id', 'predicted_effect_transcript_name',\n",
       "       'predicted_effect_original_protein_sequence',\n",
       "       'predicted_effect_aa_mutation_start_offset',\n",
       "       'predicted_effect_aa_mutation_end_offset',\n",
       "       'predicted_effect_mutant_protein_sequence', 'protein_sequence',\n",
       "       'protein_sequence_ends_with_stop_codon', 'protein_sequence_gene_names',\n",
       "       'protein_sequence_gene_ids', 'protein_sequence_transcript_names',\n",
       "       'protein_sequence_transcript_ids',\n",
       "       'protein_sequence_mutation_start_idx',\n",
       "       'protein_sequence_mutation_end_idx',\n",
       "       'trimmed_predicted_mutant_protein_sequence',\n",
       "       'trimmed_reference_protein_sequence',\n",
       "       'protein_sequence_contains_mutation',\n",
       "       'protein_sequence_contains_deletion',\n",
       "       'protein_sequence_matches_predicted_mutation_effect',\n",
       "       'filter:predicted_effect_modifies_protein_sequence',\n",
       "       'filter:has_mutant_protein_sequence_from_rna',\n",
       "       'filter:protein_sequence_contains_mutation', 'filter:min_num_alt_reads',\n",
       "       'filter:max_fraction_other_fragments', 'passes_all_filters'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s_pass."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
